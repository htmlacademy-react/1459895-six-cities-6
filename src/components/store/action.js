export const ActionType = {
  SET_CITY: `main/setCity`,
};

export const ActionCreator = {
  setCity: (city) => ({
    type: ActionType.SET_CITY,
    payload: city,
  }),
};
