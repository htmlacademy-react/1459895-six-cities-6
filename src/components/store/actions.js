export const SET_CITY = `main/setCity`;
export const SET_OPTION = `main/setOption`;
