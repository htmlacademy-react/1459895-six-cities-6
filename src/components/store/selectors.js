import {createSelector} from "reselect";

const getCity = (state) => state.city;
const getOffers = (state) => state.offers;
const getOption = (state) => state.option;
const getReviews = (state) => state.reviews;

export const getActiveOffers = createSelector(
    [getCity, getOffers, getOption],
    (city, offers, option) => {
      const activeOffers = offers.filter((offer) => offer.city.name === city);

      switch (option) {
        case `Price: low to high`:
          return activeOffers.sort((a, b) => a.price - b.price);
        case `Price: high to low`:
          return activeOffers.sort((a, b) => b.price - a.price);
        case `Top rated first`:
          return activeOffers.sort((a, b) => b.rating - a.rating);
        default:
          return activeOffers;
      }
    }
);

export const getActiveReviews = createSelector(
    [getReviews],
    (reviews) => {
      const activeReviews = reviews.sort((a, b) => new Date(b.date) - new Date(a.date));

      return activeReviews.length > 10 ? activeReviews.slice(0, 10) : activeReviews;
    }
);
