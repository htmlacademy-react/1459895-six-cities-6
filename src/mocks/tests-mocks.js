export const TEST_DATA = `test`;

export const cards = {
  start: [
    {
      id: 1,
      isFavorite: false
    },
    {
      id: 2,
      isFavorite: false
    },
    {
      id: 3,
      isFavorite: true
    }
  ],
  end: [
    {
      id: 1,
      isFavorite: false
    },
    {
      id: 2,
      isFavorite: true
    },
    {
      id: 3,
      isFavorite: true
    }
  ],
  change: {
    id: 2,
    isFavorite: true
  }
};

export const favoritesCards = {
  start: [
    {
      id: 1,
      isFavorite: true
    },
    {
      id: 2,
      isFavorite: true
    },
    {
      id: 3,
      isFavorite: true
    }
  ],
  end: [
    {
      id: 2,
      isFavorite: true
    },
    {
      id: 3,
      isFavorite: true
    }
  ],
  remove: {
    id: 1,
    isFavorite: true
  }
};

export const offers = {
  adapted: [
    {
      city: {
        name: `Paris`,
        location: {
          latitude: 48.85661,
          longitude: 2.351499,
          zoom: 13
        }
      },
      images: [
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      ],
      goods: [
        `Laptop friendly workspace`,
        `Washer`,
        `Breakfast`,
        `Air conditioning`
      ],
      host: {
        id: 25,
        name: `Angelina`,
        avatarUrl: `img/avatar-angelina.jpg`,
        isPro: true
      },
      location: {
        latitude: 48.85761,
        longitude: 2.358499,
        zoom: 16
      },
      title: `Tile House`,
      rating: 2,
      type: `house`,
      bedrooms: 2,
      price: 161,
      description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      id: 1,
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      isPremium: true,
      isFavorite: false,
      maxAdults: 9
    },
    {
      city: {
        name: `Amsterdam`,
        location: {
          latitude: 48.85661,
          longitude: 2.351499,
          zoom: 13
        }
      },
      images: [
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      ],
      goods: [
        `Laptop friendly workspace`,
        `Washer`,
        `Breakfast`,
        `Air conditioning`
      ],
      host: {
        id: 25,
        name: `Angelina`,
        avatarUrl: `img/avatar-angelina.jpg`,
        isPro: true
      },
      location: {
        latitude: 48.85761,
        longitude: 2.358499,
        zoom: 16
      },
      title: `Tile House`,
      rating: 4,
      type: `house`,
      bedrooms: 3,
      price: 161,
      description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      id: 2,
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      isPremium: true,
      isFavorite: false,
      maxAdults: 3
    }
  ],
  server: [
    {
      "city": {
        "name": `Paris`,
        "location": {
          "latitude": 48.85661,
          "longitude": 2.351499,
          "zoom": 13
        }
      },
      "images": [
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      ],
      "goods": [
        `Laptop friendly workspace`,
        `Washer`,
        `Breakfast`,
        `Air conditioning`
      ],
      "host": {
        "id": 25,
        "name": `Angelina`,
        "avatar_url": `img/avatar-angelina.jpg`,
        "is_pro": true
      },
      "location": {
        "latitude": 48.85761,
        "longitude": 2.358499,
        "zoom": 16
      },
      "title": `Tile House`,
      "rating": 2,
      "type": `house`,
      "bedrooms": 2,
      "price": 161,
      "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      "id": 1,
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      "is_premium": true,
      "is_favorite": false,
      "max_adults": 9
    },
    {
      "city": {
        "name": `Amsterdam`,
        "location": {
          "latitude": 48.85661,
          "longitude": 2.351499,
          "zoom": 13
        }
      },
      "images": [
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      ],
      "goods": [
        `Laptop friendly workspace`,
        `Washer`,
        `Breakfast`,
        `Air conditioning`
      ],
      "host": {
        "id": 25,
        "name": `Angelina`,
        "avatar_url": `img/avatar-angelina.jpg`,
        "is_pro": true
      },
      "location": {
        "latitude": 48.85761,
        "longitude": 2.358499,
        "zoom": 16
      },
      "title": `Tile House`,
      "rating": 4,
      "type": `house`,
      "bedrooms": 3,
      "price": 161,
      "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      "id": 2,
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      "is_premium": true,
      "is_favorite": false,
      "max_adults": 3
    }
  ]
};

export const reviews = {
  server: [
    {
      "user": {
        "id": 11,
        "name": `Jack`,
        "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg`,
        "is_pro": false
      },
      "id": 1,
      "rating": 3,
      "comment": `I stayed here for one night and it was an unpleasant experience.`,
      "date": `2021-02-18T15:34:01.286Z`
    },
    {
      "user": {
        "id": 11,
        "name": `Jack`,
        "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg`,
        "is_pro": false
      },
      "id": 2,
      "rating": 4,
      "comment": `The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.`,
      "date": `2021-02-18T15:34:01.286Z`
    }
  ],
  adapted: [
    {
      user: {
        id: 11,
        name: `Jack`,
        avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg`,
        isPro: false
      },
      id: 1,
      rating: 3,
      comment: `I stayed here for one night and it was an unpleasant experience.`,
      date: `2021-02-18T15:34:01.286Z`
    },
    {
      user: {
        id: 11,
        name: `Jack`,
        avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg`,
        isPro: false
      },
      id: 2,
      rating: 4,
      comment: `The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.`,
      date: `2021-02-18T15:34:01.286Z`
    }
  ]
};

export const offer = {
  adapted: {
    city: {
      name: `Paris`,
      location: {
        latitude: 48.85661,
        longitude: 2.351499,
        zoom: 13
      }
    },
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`
    ],
    goods: [
      `Laptop friendly workspace`,
      `Washer`,
      `Breakfast`,
      `Air conditioning`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatarUrl: `img/avatar-angelina.jpg`,
      isPro: true
    },
    location: {
      latitude: 48.85761,
      longitude: 2.358499,
      zoom: 16
    },
    title: `Tile House`,
    rating: 2,
    type: `house`,
    bedrooms: 2,
    price: 161,
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    id: 1,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    isPremium: true,
    isFavorite: false,
    maxAdults: 9
  },
  server: {
    "city": {
      "name": `Paris`,
      "location": {
        "latitude": 48.85661,
        "longitude": 2.351499,
        "zoom": 13
      }
    },
    "images": [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    ],
    "goods": [
      `Laptop friendly workspace`,
      `Washer`,
      `Breakfast`,
      `Air conditioning`
    ],
    "host": {
      "id": 25,
      "name": `Angelina`,
      "avatar_url": `img/avatar-angelina.jpg`,
      "is_pro": true
    },
    "location": {
      "latitude": 48.85761,
      "longitude": 2.358499,
      "zoom": 16
    },
    "title": `Tile House`,
    "rating": 2,
    "type": `house`,
    "bedrooms": 2,
    "price": 161,
    "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    "id": 1,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    "is_premium": true,
    "is_favorite": false,
    "max_adults": 9
  },
};

export const authInfo = {
  server: {
    "id": 1,
    "email": `mail@gmail.com`,
    "name": `alex`,
    "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg`,
    "is_pro": false
  },
  adapted: {
    id: 1,
    email: `mail@gmail.com`,
    name: `alex`,
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg`,
    isPro: false
  }
};

export const options = [`Popular`, `Price: low to high`, `Price: high to low`, `Top rated first`];
