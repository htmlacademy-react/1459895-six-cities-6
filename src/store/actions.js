export const SET_CITY = `main/setCity`;
export const SET_OPTION = `main/setOption`;
export const SET_AUTHORIZATION = `user/setAuthorization`;
export const SET_OFFERS = `main/setOffers`;
export const SET_NEABY_OFFERS = `proprty/setNearbyOffers`;
export const SET_REVIEWS = `proprty/setReviews`;
export const SET_FAVORITES = `favorite/setFavorites`;
export const REDIRECT_TO_ROUTE = `login/redirectToRoute`;
export const SET_OFFER = `property/setOffer`;
export const SET_IS_LOADED = `property/setIsLoaded`;
export const SET_DISABLED = `property/setDisabled`;
export const SET_IS_ERROR = `property/setIsError`;
export const UPDATE_OFFERS = `main/updateOffers`;
export const UPDATE_FAVORITES = `favorite/updateFavorites`;
export const UPDATE_OFFER = `property/updateOffer`;
export const UPDATE_NEARBY_OFFERS = `property/updateNearbyOffers`;
export const SET_IS_DATA_LOADED = `main/setIsDataLoaded`;
