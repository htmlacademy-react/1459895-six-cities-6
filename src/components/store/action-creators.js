import * as ActionType from "./actions";

export const setCity = (city) => ({type: ActionType.SET_CITY, payload: city});
export const setOption = (option) => ({type: ActionType.SET_OPTION, payload: option});
export const requireAuthorization = (status) => ({type: ActionType.REQUIRED_AUTHORIZATION, payload: status});
export const setOffers = (data) => ({type: ActionType.SET_OFFERS, payload: data});
export const setNearbyOffers = (data) => ({type: ActionType.SET_NEABY_OFFERS, payload: data});
