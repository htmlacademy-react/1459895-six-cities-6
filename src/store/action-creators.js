import * as ActionType from "./actions";
import {createAction} from '@reduxjs/toolkit';

export const setCity = createAction(ActionType.SET_CITY, (city) => {
  return {
    payload: city
  };
});
export const setOption = createAction(ActionType.SET_OPTION, (option) => {
  return {
    payload: option
  };
});
export const setAuthorization = createAction(ActionType.SET_AUTHORIZATION, (data) => {
  return {
    payload: data
  };
});
export const setOffers = createAction(ActionType.SET_OFFERS, (data) => {
  return {
    payload: data
  };
});
export const setNearbyOffers = createAction(ActionType.SET_NEABY_OFFERS, (data) => {
  return {
    payload: data
  };
});
export const setReviews = createAction(ActionType.SET_REVIEWS, (data) => {
  return {
    payload: data
  };
});
export const setFavorite = createAction(ActionType.SET_FAVORITES, (data) => {
  return {
    payload: data
  };
});
export const redirectToRoute = createAction(ActionType.REDIRECT_TO_ROUTE, (url) => {
  return {
    payload: url
  };
});
export const setOffer = createAction(ActionType.SET_OFFER, (data) => {
  return {
    payload: data
  };
});
export const setIsLoaded = createAction(ActionType.SET_IS_LOADED, (isLoaded) => {
  return {
    payload: isLoaded
  };
});
export const setDisabled = createAction(ActionType.SET_DISABLED, (isDisabled) => {
  return {
    payload: isDisabled
  };
});
export const setIsError = createAction(ActionType.SET_IS_ERROR, (isError) => {
  return {
    payload: isError
  };
});

export const updateOffers = createAction(ActionType.UPDATE_OFFERS, (data) => {
  return {
    payload: data
  };
});

export const updateFavorites = createAction(ActionType.UPDATE_FAVORITES, (data) => {
  return {
    payload: data
  };
});

export const updateOffer = createAction(ActionType.UPDATE_OFFER, (data) => {
  return {
    payload: data
  };
});

export const updateNearbyOffers = createAction(ActionType.UPDATE_NEARBY_OFFERS, (data) => {
  return {
    payload: data
  };
});

export const setIsDataLoaded = createAction(ActionType.SET_IS_DATA_LOADED, (isLoad) => {
  return {
    payload: isLoad
  };
});
