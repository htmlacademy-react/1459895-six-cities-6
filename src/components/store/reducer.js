import {ActionType} from "./action";
import {offers} from "../../mocks/offers";

const initialState = {
  city: `Paris`,
  offers: [...offers],
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_CITY:
      return {
        ...state,
        city: action.payload
      };
    default:
      return state;
  }
};

export {reducer};
