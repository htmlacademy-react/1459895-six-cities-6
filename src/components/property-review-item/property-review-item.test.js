import React from 'react';
import {render} from '@testing-library/react';
import PropertyReviewItem from './property-review-item';
import {reviews} from "../../store/tests-mocks";

describe(`Test PropertyReviewItem component`, () => {
  it(`PropertyReviewItem component should render correctly`, () => {
    const {container} = render(
        <PropertyReviewItem
          review={reviews.adapted[0]}
        />
    );
    expect(container).toMatchSnapshot();
  });
});
