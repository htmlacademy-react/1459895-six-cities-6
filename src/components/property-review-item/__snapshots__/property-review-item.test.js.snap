// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PropertyReviewItem component PropertyReviewItem component should render correctly 1`] = `
<div>
  <li
    class="reviews__item"
  >
    <div
      class="reviews__user user"
    >
      <div
        class="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          class="reviews__avatar user__avatar"
          height="54"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg"
          width="54"
        />
      </div>
      <span
        class="reviews__user-name"
      >
        Jack
      </span>
    </div>
    <div
      class="reviews__info"
    >
      <div
        class="reviews__rating rating"
      >
        <div
          class="reviews__stars rating__stars"
        >
          <span
            style="width: 60%;"
          />
          <span
            class="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        class="reviews__text"
      >
        I stayed here for one night and it was an unpleasant experience.
      </p>
      <time
        class="reviews__time"
        datetime="2019-04-24"
      >
        February 2021
      </time>
    </div>
  </li>
</div>
`;
